examples/advanced/demo_optimizing_physics_parameter.py

```diff
@@ -0,0 +1,258 @@
+    rho = ind[cond][-1]
+    theta = cssv[cond][-1] / rho
+    w = torch.maximum(v - theta, torch.zeros_like(v))
+    return w.reshape(shape)
+
+
+# We also define a data fidelity term
+data_fidelity = dinv.optim.L2()
+# %%
+#
+# Run the algorithm
+#
+# Initialize a constant kernel
+kernel_init = torch.zeros_like(true_kernel)
+kernel_init[..., 5:-5, 5:-5] = 1.0
+kernel_init = projection_simplex_sort(kernel_init)
+n_iter = 1000
+stepsize = 0.7
+
+kernel_hat = kernel_init
+losses = []
+for i in tqdm(range(n_iter)):
+    # compute the gradient
+    with torch.enable_grad():
+        kernel_hat.requires_grad_(True)
+        physics.update(filter=kernel_hat)
+        loss = data_fidelity(y=y, x=x, physics=physics) / y.numel()
+        loss.backward()
+    grad = kernel_hat.grad
+
+    # gradient step and projection step
+    with torch.no_grad():
+        kernel_hat = kernel_hat - stepsize * grad
+        kernel_hat = projection_simplex_sort(kernel_hat)
+
+    losses.append(loss.item())
+
+dinv.utils.plot(
+    [true_kernel, kernel_init, kernel_hat],
+    titles=["True kernel", "Init. kernel", "Estimated kernel"],
+    suptitle="Result with Projected Gradient Descent",
+)
+
+# %%
+#
+# We can plot the loss to make sure that it decreases
+#
+plt.figure()
+plt.plot(range(n_iter), losses)
+plt.title("Loss evolution")
+plt.yscale("log")
+plt.xlabel("Iteration")
+plt.tight_layout()
+plt.show()
+
+# %%
+# Combine with arbitrary optimizer
+# --------------------------------
+#
+# Pytorch provides a wide range of optimizer for training neural networks.
+# We can also pick one of those to optimizer our parameter
+
+kernel_init = torch.zeros_like(true_kernel)
+kernel_init[..., 5:-5, 5:-5] = 1.0
+kernel_init = projection_simplex_sort(kernel_init)
+
+kernel_hat = kernel_init.clone()
+optimizer = torch.optim.Adam([kernel_hat], lr=0.1)
+
+# We will alternate a gradient step and a projection step
+losses = []
+n_iter = 200
+for i in tqdm(range(n_iter)):
+    optimizer.zero_grad()
+    # compute the gradient, this will directly change the gradient of `kernel_hat`
+    with torch.enable_grad():
+        kernel_hat.requires_grad_(True)
+        physics.update(filter=kernel_hat)
+        loss = data_fidelity(y=y, x=x, physics=physics) / y.numel()
+        loss.backward()
+
+    # a gradient step
+    optimizer.step()
+    # projection step, when doing additional steps, it's important to change only
+    # the tensor data to avoid breaking the gradient computation
+    kernel_hat.data = projection_simplex_sort(kernel_hat.data)
+    # loss
+    losses.append(loss.item())
+
+dinv.utils.plot(
+    [true_kernel, kernel_init, kernel_hat],
+    titles=["True kernel", "Init. kernel", "Estimated kernel"],
+    suptitle="Result with ADAM",
+)
+
+# %%
+#
+# We can plot the loss to make sure that it decreases
+#
+plt.figure()
+plt.plot(range(n_iter), losses)
+plt.title("Loss evolution")
+plt.yscale("log")
+plt.xlabel("Iteration")
+plt.tight_layout()
+plt.show()
```

Comment:
Using `plt.semilogy` can make you gain 1 line

```suggestion
plt.figure()
plt.semilogy(range(n_iter), losses)
plt.title("Loss evolution")
plt.xlabel("Iteration")
plt.tight_layout()
plt.show()
```